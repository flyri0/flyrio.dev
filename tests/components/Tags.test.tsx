import { cleanup, render, screen } from '@testing-library/react';

import Tags from '@components/Tags';

describe('Tags', () => {
  const THREE_TAGS = ['Java', 'Tetris', 'Rage'];
  const FOUR_TAGS = ['Arch Linux', 'ILoveCandy', 'ParallelDownloads', '5'];

  test('renders correctly', () => {
    const component = render(<Tags tags={THREE_TAGS} />);
    expect(component).toMatchSnapshot();
  });

  test('render the correct number of spans', () => {
    render(<Tags tags={THREE_TAGS} />);
    expect(screen.getAllByTestId('tag').length).toBe(3);
    cleanup();

    render(<Tags tags={FOUR_TAGS} />);
    expect(screen.getAllByTestId('tag').length).toBe(4);
  });
});
