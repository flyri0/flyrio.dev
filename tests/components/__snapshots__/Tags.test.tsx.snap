// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Tags > renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex items-start gap-x-1 text-xs font-bold"
      >
        <span
          class="rounded border border-shadow p-1 text-xs text-shadow"
          data-testid="tag"
        >
          Java
        </span>
        <span
          class="rounded border border-shadow p-1 text-xs text-shadow"
          data-testid="tag"
        >
          Tetris
        </span>
        <span
          class="rounded border border-shadow p-1 text-xs text-shadow"
          data-testid="tag"
        >
          Rage
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex items-start gap-x-1 text-xs font-bold"
    >
      <span
        class="rounded border border-shadow p-1 text-xs text-shadow"
        data-testid="tag"
      >
        Java
      </span>
      <span
        class="rounded border border-shadow p-1 text-xs text-shadow"
        data-testid="tag"
      >
        Tetris
      </span>
      <span
        class="rounded border border-shadow p-1 text-xs text-shadow"
        data-testid="tag"
      >
        Rage
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
